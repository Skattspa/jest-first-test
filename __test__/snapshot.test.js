import {getCharacter} from '../snapshot';
import rick from '../rick.json';

describe('Es hora de los snapshots o instantáneas',()=>{
    test('Snapshot',()=>{
        expect(getCharacter(rick)).toMatchSnapshot();
    });

   /*  test('Siempre fallará la instantánea',()=>{
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20)
        }
        expect(user).toMatchSnapshot({
            id: expect.any(Number),
            createAt :expect.any(Date)
        });
    }); */
    test('Tenemos una excepción en el código',()=>{
        const user = {
            id: Math.floor(Math.random() * 20),
            name: "Juan M"
        }

        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        })
    });
});
