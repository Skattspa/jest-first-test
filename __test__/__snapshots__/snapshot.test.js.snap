// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de los snapshots o instantáneas Siempre fallará la instantánea 1`] = `
Object {
  "createAt": 2021-06-04T23:57:46.727Z,
  "id": 11,
}
`;

exports[`Es hora de los snapshots o instantáneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de los snapshots o instantáneas Tenemos una excepción en el código 1`] = `
Object {
  "id": Any<Number>,
  "name": "Juan M",
}
`;
